### PR TITLE
[TAS-1947] 📈 Fix ecommerce event in carts and support quantity

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

File: src/util/EventLogger.js [hunk 1]
```diff
@@ -94,14 +95,17 @@ export function logPurchaseFlowEvent(
...
-        items: items.map(i => ({
-          item_id: i.classId,
-          item_name: i.name?.substring(0, 100),
-          item_brand: isNFTBook ? 'Book NFT' : 'Writing NFT',
-          currency,
-          price: i.price,
-          quantity: 1,
-        })),
+        items: items.map(i => {
+          const itemId = i.productId || i.collectionId || i.collectionId;
```
Comment (nwingt) [inline src/util/EventLogger.js:99]:
```suggestion
          const itemId = i.productId || i.collectionId;
```

File: src/pages/shopping-cart/wnft.vue [hunk 2]
```diff
@@ -207,6 +207,7 @@ export default {
   mounted() {
     this.updateLIKEPrice();
     logPurchaseFlowEvent(this, 'add_to_cart', this.purchaseEventParams);
```
Comment (nwingt) [inline src/pages/shopping-cart/wnft.vue:209]: Should we keep triggering `add_to_cart` when entering this page?

File: src/pages/nft/collection/_collectionId/index.vue [hunk 3]
```diff
@@ -287,20 +298,29 @@ export default {
...
-                price: this.collectionPrice,
+                price: customPriceInDecimal || this.collectionPrice,
                 collection: this.collectionId,
               },
             ],
-            price: this.collectionPrice,
+            price: totalPrice,
             currency: 'USD',
             isNFTBook: true,
           };
```
Comment (nwingt) [inline src/pages/nft/collection/_collectionId/index.vue:317]: Is this replaced by `this.purchaseEventParams`?

File: src/pages/shopping-cart/wnft.vue (hunk 2 above)
Comment (williamchong) [reply src/pages/shopping-cart/wnft.vue:209]: In wnft I think the user would always go into cart page when clicking `purchase all`?
But wnft flow is not very important now i guess

File: src/pages/nft/collection/_collectionId/index.vue (hunk 3 above)
Comment (williamchong) [reply src/pages/nft/collection/_collectionId/index.vue:317]: Yes

**Review by nwingt — APPROVED** [review]

**Pull request merged**